internal/pkg/dynamic-config/dynamic_context.go

```diff
@@ -31,11 +32,18 @@ func NewDynamicContext(context *v1.Context, v2Client *ccloudv2.Client) *DynamicC
 	}
 }
 
-func (d *DynamicContext) ParseFlagsIntoContext(cmd *cobra.Command) error {
+func (d *DynamicContext) ParseFlagsIntoContext(cmd *cobra.Command, isTest bool) error {
 	if environment, _ := cmd.Flags().GetString("environment"); environment != "" {
 		if d.GetCredentialType() == v1.APIKey {
 			output.ErrPrintln("WARNING: The `--environment` flag is ignored when using API key credentials.")
 		} else {
+			if !isTest {
```

Comment:
Why are we choosing not to test this?

Reply:
This just prevents the `dynamic_context_test.go` unit test from calling `cluster` and `environment` list APIs, which will cause the test to fail because the mocker is not defined in the SDK. However, these blocks of code are still tested in integration tests.

Reply:
Should we add those mocks? Better yet, can we delete those tests and replace them with integration tests in a separate PR?

Reply:
I think we should be able to remove this test: https://github.com/confluentinc/cli/blob/e5263f122489f5937216cfc44aba58ecbc658dfb/internal/pkg/dynamic-config/dynamic_context_test.go#L85
Since the functionality of `ParseFlagsIntoContext` is already being relied on in integration tests

Reply:
Let's try to get full coverage of this function with our integration tests... and if it can be written in a way that's easier to unit test, even better!